server/service/src/invoice_line/stock_in_line/mod.rs

```diff
@@ -11,12 +12,68 @@ pub use self::update::*;
 pub enum StockInType {
     #[default]
     InboundReturn,
+    InventoryAddition,
 }
 
 impl StockInType {
     pub fn to_domain(&self) -> InvoiceRowType {
         match self {
             StockInType::InboundReturn => InvoiceRowType::InboundReturn,
+            StockInType::InventoryAddition => InvoiceRowType::InventoryAddition,
         }
     }
 }
+
+pub struct StockLineInput {
+    pub store_id: String,
+    pub on_hold: bool,
+    pub barcode_id: Option<String>,
+    pub supplier_link_id: String,
+}
+
+pub fn generate_batch(
```

Comment:
Added a couple of fields (on_hold, barcode_id) and realised stock_in_line was actually using the `generate_batch` from inbound_shipment_line. Didn't want to touch inbound shipment, so created a `generate_batch` for stock_in_line.

Reply:
Did we end up having an issue to consolidate insert lines in inbound shipments and maybe in inventory adjustments in stocktake ?

Reply:
No issues, just haven't refactored yet... didn't want to make changes to those areas in this PR